core/src/main/scripts/importer/validateData.py

```diff
@@ -1228,6 +1232,409 @@ def checkLine(self, data):
+                        self.logger.error(log_message, extra={'line_number': self.line_number,
+                                                              'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1, 
+                                                                                         tumor_seq_allele2)})
+        return True
+
+    def checkAlleleSpecialCases(self, data):
+        """ Check other special cases which should or should not occur in Allele Based columns
+        Special cases are either from unofficial vcf2maf rules or discrepancies identified. """
+
+        # First check if columns necessary exist in the data
+        necessary_columns = ['Reference_Allele', 'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'Variant_Type']
+        if set(necessary_columns).issubset(self.cols):
+            ref_allele = data[self.cols.index('Reference_Allele')].strip()
+            tumor_seq_allele1 = data[self.cols.index('Tumor_Seq_Allele1')].strip()
+            tumor_seq_allele2 = data[self.cols.index('Tumor_Seq_Allele2')].strip()
+            variant_type = data[self.cols.index('Variant_Type')].strip()
+
+            # Check if Allele Based columns are not all the same
+            if ref_allele == tumor_seq_allele1 and tumor_seq_allele1 == tumor_seq_allele2:
+                log_message = "All Values in columns Reference_Allele, Tumor_Seq_Allele1 " \
+                              "and Tumor_Seq_Allele2 are equal."
+                if not self.strict_mode:
+                    self.logger.warning(log_message, extra={'line_number': self.line_number,
+                                                            'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1,
+                                                                                       tumor_seq_allele2)})
+                else:
+                    self.logger.error(log_message, extra={'line_number': self.line_number,
+                                                          'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1, 
+                                                                                     tumor_seq_allele2)})
+
+            # Check in case of insertion if the Reference_Allele contains '-'
+            if variant_type == "INS" and '-' not in ref_allele:
+                log_message = "Variant_Type indicates an insertion, but Reference_Allele does not equal -."
+                if not self.strict_mode:
+                    self.logger.warning(log_message, extra={'line_number': self.line_number, 'cause': ref_allele})
+                else:
+                    self.logger.error(log_message, extra={'line_number': self.line_number, 'cause': ref_allele})
+
+            # In case of deletion, check when Reference_Allele is the same length as both Tumor_Seq_Allele if at least
+            # one of the Tumor_Seq_Alleles is a deletion ('-') otherwise a SNP
+            if variant_type == "DEL" and len(ref_allele) == len(tumor_seq_allele1) \
```

Comment:
Similar question as above. What about GT -> A ?

Reply:
In this case, it is first checking if the reference and tumor seq alleles are the same sizes. In case they are, there should be '-' in the tumor seq alleles, otherwise it indicates an SNP.

(check if length reference allele is bigger than tumor seq alleles is done in MAF file check 11)